### ISSUE TITLE
indent_constr_colon not working on line continuation

### ISSUE THREAD

**Issue by soeugene** [uncrustify/uncrustify#1159]
uncrustify 0.64

Given the file:
```
namespace ns1
{
Class1::Class1(boost::shared_ptr<messaging::IReceive> receiver, boost::shared_ptr<messaging::ITransmit> transmitter) :
    m_a(""),
    m_b(""),
    m_c(receiver),
    m_d(transmitter),
    m_d(true)
{

}

Class1::Class1(boost::shared_ptr<messaging::IReceive> receiver,
    boost::shared_ptr<messaging::ITransmit> transmitter) :
        m_a(""),
        m_b(""),
        m_c(receiver),
        m_d(transmitter),
        m_d(true)
{

}

}
```

uncrustify is not properly handling indentation of the class initializers in the second block where the class definition is split across two lines.  I expect the code to be unchanged by uncrustify.  Using the command line:

```
uncrustify -p debug.txt -c .uncrustifyrc -f uncrustify_test.txt -o uncrustify_test.new -l CPP
```

[debug.txt](https://github.com/uncrustify/uncrustify/files/1001727/debug.txt)

[//]: # "  If the issue is connected to existing Uncrustify options please, if possible, add the"
[//]: # "  following information to ease up the process:"
[//]: # "  • a link to a debug file:"
[//]: # "      generated with: 'uncrustify -p debug.txt -c pathToUsedConfig.cfg -f toBeFormatedFile.cpp' "
[//]: # "      Example: [debug.txt](https://linkToTheFile)"
[//]: # "      Example hosters for debug files: pastbin.com, gist.github.com, ..."
[//]: # "      The used config file is included in the debug file and does not need to be included here."
[//]: # 
[//]: # "  • include a small but complete test file that will be uncrustifyed"
[//]: # "  • include the generated results"
[//]: # "  • include the expected results"
[//]: # 
[//]: # "  ✋ please add a line containing ``` above and below of each of those three code sections"
[//]: # 
[//]: # "  • include the current version of your Uncrustify executable"
[//]: # "      printout via 'uncrustify -v'"
[//]: # "      Example: current version: uncrustify 0.63"
[//]: # "      or if possible additionally with the git sha of the commit"
[//]: # "                    current version: uncrustify 0.63 dc7b412"
[//]: # 
[//]: # "  • if possible include a version that worked"
[//]: # "      Example: working version: uncrustify 0.63"
[//]: # "      or"
[//]: # "                    working version: uncrustify 0.63 2a5e88f"

**Comment by guy-maurel** [comment]
uncrustify is running correctly.
The indentation in the second block, for the second line `boost::shared_ptr<m...` is controlled by:
`indent_continue  = 4`
All the rest is controlled by:
`indent_columns = 4`
This is not a bug.

**Comment by soeugene** [comment]
Shouldn't it recognize the class initializer colon on the line `boost::shared_ptr<messaging::ITransmit> transmitter) :` 
and indent the following line as per indent_constr_colon?

**Comment by guy-maurel** [comment]
With the config:
```
input_tab_size      = 4
output_tab_size     = input_tab_size
indent_constr_colon = true
nl_constr_init_args = force
nl_constr_colon     = force
pos_constr_comma    = trail_force
pos_constr_colon    = lead_force
indent_columns      = 4
indent_continue     = 4
```
You get:
```
namespace ns1
{

Class1::Class1(boost::shared_ptr<messaging::IReceive> receiver, boost::shared_ptr<messaging::ITransmit> transmitter)
        : m_a(""),
          m_b(""),
          m_c(receiver),
          m_d(transmitter),
          m_d(true)
{

}
Class1::Class1(boost::shared_ptr<messaging::IReceive> receiver,
    boost::shared_ptr<messaging::ITransmit> transmitter)
        : m_a(""),
          m_b(""),
          m_c(receiver),
          m_d(transmitter),
          m_d(true)
{
}

}
```
Please test and report.

**Comment by soeugene** [comment]
Please explain why I need the configuration options:
input_tab_size      = 4
output_tab_size     = input_tab_size
nl_constr_init_args = force
nl_constr_colon     = force
pos_constr_comma    = trail_force
pos_constr_colon    = lead_force
It still did not solve all the  problems for me.

[debug.txt](https://github.com/uncrustify/uncrustify/files/1447823/debug.txt)
[uncrustify_test2.txt](https://github.com/uncrustify/uncrustify/files/1447824/uncrustify_test2.txt)

**Comment by soeugene** [comment]
Also why can't the constructor initializer colon remain on the same line as per the original file?

**Comment by guy-maurel** [comment]
Yes, the constructor initializer colon remain on the same line by using the option
`pos_constr_colon    = trail_force`
There is no option to indent more than configured with
`indent_columns      = 4`
as a continuation line doesn't change/increment the indentation level.

**Comment by soeugene** [comment]
Maybe what I really need is an option to allow multiple indentation levels on continuation lines.  Is that something you could add?

**Comment by guy-maurel** [comment]
Please give an example of what you are looking for:
input file, expected output

**Comment by mwoehlke-kitware** [comment]
I think what @soeugene wants is for the indent of the member initializers to be relative to the indent of the line containing the `:`.

That's... atypical, in my experience, though not entirely unreasonable.
